### PR TITLE
jclklib: Add composite event support

### PR DESCRIPTION
**Pull request by yoongsiang2**
Allow user to set composite event where they will only be notified whether all events are meet or at least one of the event is not meet.

### CODE REVIEW

**Comment by yoongsiang2** [comment]
image

File: jclklib/client/notification_msg.cpp [hunk 1]
```diff
@@ -109,10 +113,39 @@ PROCESS_MESSAGE_TYPE(ClientNotificationMessage::processMessage)
 		client_ptp_data.gmPresent_event_count.fetch_add(1, std::memory_order_relaxed);
 	}
 
+	if (composite_eventSub[0]) {
+		old_composite_event = composite_client_ptp_data.composite_event;
```
Comment (AzuraTarmiziIntel) [inline jclklib/client/notification_msg.cpp:117]: Just a note. maybe if this composite event test is reused again elsewhere, we can consider to make a small helper function.

**Comment by yoongsiang2** [comment]
new look of printing table:
image

File: jclklib/client/notification_msg.cpp (hunk 1 above)
Comment (yoongsiang2) [reply jclklib/client/notification_msg.cpp:117]: yes, you are right. we plan to refactor parse_buffer and process_message function of subscribe and notify message in next PR.

**Review by AzuraTarmiziIntel — APPROVED** [review]
LGTM

**Pull request merged**